x-pack/plugins/security/public/lib/kibana_privilege_calculator/kibana_allowed_privileges_calculator.ts

```diff
@@ -93,6 +93,10 @@ export class KibanaAllowedPrivilegesCalculator {
     candidateFeaturePrivileges: string[]
   ): { privileges: string[]; canUnassign: boolean } {
     const effectiveFeaturePrivilegeExplanation = effectivePrivileges.feature[featureId];
+    if (effectiveFeaturePrivilegeExplanation == null) {
+      throw new Error('To calculate allowed feature privileges, we need the effective privileges');
```

Comment:
This code-path shouldn't be hit because the allowed privileges require the effective privileges to calculate, but the explicit Error should help us if this is no longer the case.